### DISCUSSION TITLE
TRPCClient Error: Unexpected end of JSON input

### DISCUSSION THREAD

**Discussion by franfares10** [trpc/trpc#5478] [Q&A]
Hi everyone!

I'm having troubles with my app, I want to make a typical form mutation, and I'm getting `405 Method Not Allowed`.
The curious thing is that it works correctly on local environment but it crashes on prod, I hosted the app in Vercel.

<img width="891" alt="Screenshot 2024-02-13 at 13 03 38" src="https://github.com/trpc/trpc/assets/59488971/d4667bb0-135c-42ea-8940-72236002e097">

**Comment by anis-dr** [comment]
I have the same problem. Did you find a solution please?

**Comment by KATT** [comment]
Y'all need to provide more context if you want help

**Reply by spaceemotion** [reply]
I am frequently getting the same error, but there's actually not a lot of context to be had. i personally never have that error locally, or even on the live service. it seems to happen to people with less stable internet connections though?

Here is a screengrab from sentry on this, as you can see it's actually missing most of the context: 
![spaceemotion sentry io_share_issue_316c93df11d941c1907b0d7d277bab76_](https://github.com/trpc/trpc/assets/429147/430ee3a4-ce81-4126-bdff-5d1922273f15)

When I dig deeper into the event details it seems it got back a 401 from the server, but maybe the body was empty? (this was on an android phone, so yeah, internet could have been very spotty!).

**Comment by DiegoAraujoJS** [comment]
Same issue here, up to now I've seen it happening only on mutations, never on queries. The behaviour is pretty random. A temporal fix for me was to change the methods from mutation to query.

**Reply by matteoolefloch** [reply]
The issue on my side appeared somewhat randomly on Chrome for certain mutations, but on Safari, it occurred with every mutation.

For reference, I have upgraded TRPC to the latest version, `11.0.0-rc.439`, which resolved the problem. Previously, I was using version `11.0.0-rc.364`.
